src/shogun/multiclass/KNN.cpp

```diff
@@ -237,12 +242,17 @@ CMulticlassLabels* CKNN::classify_NN()
 
 SGMatrix<int32_t> CKNN::classify_for_multiple_k()
 {
-	ASSERT(m_num_classes>0)
-	ASSERT(distance)
-	ASSERT(distance->get_num_vec_rhs())
+	REQUIRE(distance, "No distance assigned.\n");
+	REQUIRE(
+	    m_num_classes > 0,
+	    "Number of classes is zero. You may need to call train()\n");
 
 	int32_t num_lab=distance->get_num_vec_rhs();
-	ASSERT(m_k<=num_lab)
+	REQUIRE(num_lab, "No vectors on right hand side\n");
+
+	REQUIRE(
+	    m_k <= num_lab, "Number of labels (%d) is less than K (%d).\n", num_lab,
```

Comment:
Say what is expected rather than what you got
"Number of labels (%d) must be at least K (%d)"